### ISSUE TITLE
[a11y]: The explicitly-assigned ARIA role "list" is redundant with the implicit role of the element <ul>

### ISSUE THREAD

**Issue by dishukataria** [carbon-design-system/carbon#22432]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

v2.66.0

### React version

v18.3.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The ContainedList component includes an explicit role="list" on an HTML `<ul>` element, which already has an implicit ARIA role of "list." This redundant role declaration violates best practices for ARIA.

<img width="1728" height="950" alt="Image" src="https://github.com/user-attachments/assets/0e8b53b1-5757-4e2d-aa85-0aa0bbb1723c" />

### Reproduction/example

https://stackblitz.com/edit/vitejs-vite-doankyac?file=src%2FApp.tsx

### Steps to reproduce

1. Use the ContainedList component (or Carbon’s wrapper) in the form:
`<ProductiveCard title={'LIST_LABEL'}>`
  `<ContainedList kind="on-page" size="xl">`
    `<ContainedListItem>…</ContainedListItem>`
  `</ContainedList>`
`</ProductiveCard>`

2. Open browser dev tools and examine the `<ul> `element output by ContainedList.
You'll observe that it has role="list", despite `<ul>` elements already providing that role implicitly via native semantics. 
3. Run an accessibility checker. It should flag a violation : _The explicitly-assigned ARIA role "list" is redundant with the implicit role of the element `<ul>`_

4. As per WAI‑ARIA best practices, adding ARIA roles that duplicate native element semantics (like role="list" on `<ul>`) is discouraged, it's redundant markup and should be removed

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by kennylam** [comment]
If this makes more sense in the Products repo, feel free to move the issue there.

**Comment by maradwan26** [comment]
The added "list" role is a remedy to a behaviour in the Safari browser added way back in https://github.com/carbon-design-system/carbon/pull/14019. Safari will not announce the list semantics for this component otherwise, which looks to be an intentional behaviour: https://bugs.webkit.org/show_bug.cgi?id=170179. The redundant role looks to be the recommendation here.
